roles/default/kiali-deploy/tasks/main.yml

```diff
@@ -355,6 +355,30 @@
   - is_openshift == True
   - kiali_vars.external_services.tracing.auth.ca_file is not defined or kiali_vars.external_services.tracing.auth.ca_file == ''
 
+# If the CR specifies Kiali's external URL, annotate the Kiali service with that url
+# This enables cross-linking between Kiali instances in multi-cluster setups.
+- name: Add external-url annotation to service
+  set_fact:
+    kiali_vars: |
+      {{ kiali_vars | combine({
```

Comment:
I think we need a molecule test for this. Have you tested to see what would happen if the user already set `spec.deployment.service_annotations` with their own custom annotations?

While this seems like a clever solution, this is going to put this value in the ConfigMap that will look different than the value the user specified in the CR. This is not entirely bad, but it will be confusing. We are essentially adding an annotation to their configuration which the user did not add.

I think a better way to do this would be to put this directly in the Service templates
* https://github.com/kiali/kiali-operator/blob/master/roles/default/kiali-deploy/templates/openshift/service.yaml#L15
* https://github.com/kiali/kiali-operator/blob/master/roles/default/kiali-deploy/templates/kubernetes/service.yaml#L14

Something like:

```
  annotations:
    service.beta.openshift.io/serving-cert-secret-name: kiali-cert-secret
    kiali.io/api-spec: https://kiali.io/api
    kiali.io/api-type: rest
{% if kiali_vars.server.web_fqdn|length > 0 and kiali_vars.server.web_schema|length > 0 %}
    kiali.io/external-url: {{ kiali_vars.server.web_schema + '://' + kiali_vars.server.web_fqdn + ((':' + kiali_vars.server.web_port | string) if (kiali_vars.server.web_port | string | length != 0) else '') + (kiali_vars.server.web_root | default(''))
{% endif %}
{% if kiali_vars.deployment.service_annotations|length > 0 %}
    {{ kiali_vars.deployment.service_annotations | to_nice_yaml(indent=0) | trim | indent(4) }}
{% endif %}
```

Reply:
ouch! Yes, your solution is much cleaner.
I'll change this PR.